### PR TITLE
Add glue catalog loader

### PR DESCRIPTION
**Pull request by ravediamond**
Thank you for contributing to LangChain!

- [ ] **Add Glue Catalog loader**: "community: Load AWS Glue Catalog table schema"

- [ ] **PR message**: 
    - **Description:** Added a loader like the AWS Athena loader to get the the schema, Pandas style, of the all or a subset of the tables inside a Glue database.
    - **Twitter handle:** RSomaw

- [ ] **Add tests and docs**: 
    - **Docs:** A doc on how to use the loader has been added.
    - **Unit tests:** Unit tests is not possible unless I add moto which by itself add a lot of dependencies. As I did not see any unit tests for other AWS related loader, I did not add them.

### CODE REVIEW

File: libs/community/langchain_community/document_loaders/glue_catalog.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    AWS credentials are automatically loaded using boto3, following the standard AWS method:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+
+    If a specific AWS profile is required, it can be specified and will be used to establish the session.
+    """
+
+    def __init__(
+        self,
+        database: str,
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/glue_catalog.py:22]:
```suggestion
        database: str,
        *,
```

File: libs/community/langchain_community/document_loaders/glue_catalog.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    This loader fetches the schema of each table within a specified AWS Glue database.
+    The schema details include column names and their data types, similar to pandas dtype representation.
+
+    AWS credentials are automatically loaded using boto3, following the standard AWS method:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+
+    If a specific AWS profile is required, it can be specified and will be used to establish the session.
+    """
+
+    def __init__(
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/glue_catalog.py:20]: Consider allowing a user to pass in a boto3 session instance directly

**Review by eyurtsev — APPROVED** [review]

File: libs/community/langchain_community/document_loaders/glue_catalog.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        database: str,
+        profile_name: Optional[str] = None,
+        table_filter: Optional[List[str]] = None,
+    ):
+        """Initialize Glue database loader.
+
+        Args:
+            database: The name of the Glue database from which to load table schemas.
+            profile_name: Optional. The name of the AWS profile to use for credentials.
+            table_filter: Optional. The list of the table to to filter.
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/glue_catalog.py:31]: What does it mean to have a list of tables to filter? Maybe paraphrase the doc-string?

```suggestion
            table_filter: Optional. The list of the table to filter.
```

**Comment by eyurtsev** [comment]
This PR is ready to merge if it passes unit tests.

@ravediamond could you confirm whether you want or do not want to update init to support passing a session in.

File: libs/community/langchain_community/document_loaders/glue_catalog.py (hunk 2 above)
Comment (ravediamond) [reply libs/community/langchain_community/document_loaders/glue_catalog.py:22]: Nice idea. I just did it.

File: libs/community/langchain_community/document_loaders/glue_catalog.py (hunk 1 above)
Comment (ravediamond) [reply libs/community/langchain_community/document_loaders/glue_catalog.py:24]: My bad. Fixed it.

**Review by ravediamond — COMMENTED** [review]
I fixed your comments and also added better documentation for the `table_filter` argument.

File: libs/community/langchain_community/document_loaders/glue_catalog.py (hunk 3 above)
Comment (ravediamond) [reply libs/community/langchain_community/document_loaders/glue_catalog.py]: You are right, it was not clear. I rewrited this part to be more explicit.

**Comment by ravediamond** [comment]
@eyurtsev I added the boto session argument and update the doc. It is ready to be merged.

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
linted the code + added TYPE_CHECKING

**Pull request merged**